test/Common/Program.cs

```diff
@@ -66,7 +66,7 @@ private static int ShowSdkInfo()
     {
         var log = new StringTestLogger();
         var command = new DotnetCommand(log, "--info");
-        var testDirectory = TestDirectory.Create(Path.Combine(TestContext.Current.TestExecutionDirectory, "sdkinfo"));
+        var testDirectory = TestDirectory.Create(Path.Combine(TestContext.Current?.TestExecutionDirectory!, "sdkinfo"));
```

Comment:
I wonder if rather than having the ! here we should have the property getter for TestExecutionDirectory throw if it's null (as it really should never be null). That would potentially remove a lot of these !s that you've added by doing it in one place.